### ISSUE TITLE
multiline modloadmsg has undesired indentation

### ISSUE THREAD

**Issue by SethosII** [easybuilders/easybuild-framework#2120]
At the begining of each line after the first there are spaces that aren't part of `modloadmsg`. For example when loading Trimmomatic:

```
user@node:~$ module load Trimmomatic
To execute Trimmomatic run: java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.36.jar
    user@node:~$
```

The easyconfig contains `modloadmsg = """To execute Trimmomatic run: java -jar $EBROOTTRIMMOMATIC/trimmomatic-%(version)s.jar\n"""`, so there is no space after the newline.

The generated module file contains this:

```
if mode() == "load" then
    io.stderr:write([==[To execute Trimmomatic run: java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.36.jar
    ]==])
end
```

I think the indentation is inherited from python formatting?

EasyBuild version 3.1.0

**Comment by damianam** [comment]
In addition to that, wouldn't it make sense to use `LmodMessage` instead of `io.stderr:write`?

**Comment by boegel** [comment]
@SethosII Yes, you're probably right, although it's a bit hard to see that in the code straight away.

See `msg_on_load` in `module_generator.py` (https://github.com/hpcugent/easybuild-framework/blob/master/easybuild/tools/module_generator.py#L674), the message is being inserted with the newline included, but then the whole thing gets indented because it's wrapper in `self. conditional_statement`.

So, it's not so much about Python indentation, but because of the `mode() == "load"` guard...

I'm not sure how to fix this, since this may also affect messages that have multiple newlines in them... Do these get this weird indentation as well?

@damianam Using `LmodMessage` may make sense, but is this tied to a paticular minimal Lmod version (that is newer than what we require now, i.e. 5.8?

**Comment by damianam** [comment]
This is the first reference I see of `LmodMessage`: https://github.com/TACC/Lmod/commit/5d4166b26958b25279c1b1b3b82ee99c7f90a7e9 That's from `3.2`

**Comment by SethosII** [comment]
@boegel Here is an example of a `modloadmsg` with more lines:

```
# modloadmsg = """line1\nline2\n\n    line3\nline4\n"""
user@node$ module load multilinetest
line1
    line2
    
        line3
    line4
    user@node$
```

So the first line has no indentation but all subsequent lines and intentional indentation is preserved.

**Comment by SethosII** [comment]
I just noticed that it works as expected in Easybuild `3.8.0`:

```
# modloadmsg = """line1\nline2\n\n    line3\nline4\n"""
user@node$ module load multilinetest
line1
line2

    line3
line4
user@node$
```

Also the newline character at the end of the string doesn't seem to be necessary anymore:

```
# modloadmsg = """line1\nline2\n\n    line3\nline4"""
user@node$ module load multilinetest
line1
line2

    line3
line4
user@node$
```

**Comment by boegel** [comment]
@SethosII Indeed, fixed in #2613 (EasyBuild v3.7.1)
